runtimes/js/src/api.rs

```diff
@@ -87,15 +88,36 @@ impl Request {
 pub struct APIPromiseHandler;
 
 impl PromiseHandler for APIPromiseHandler {
-    type Output = Result<schema::JSONPayload, api::Error>;
+    type Output = HandlerResponse;
 
     fn resolve(&self, env: Env, val: Option<napi::JsUnknown>) -> Self::Output {
         let Some(val) = val else {
-            return Ok(None);
+            return Ok(HandlerResponseInner {
+                payload: None,
+                extra_headers: None,
+            });
         };
 
-        match parse_pvalues(val) {
-            Ok(val) => Ok(val),
+        let obj: JsObject = val
+            .try_into()
+            .map_err(|e| api::Error::invalid_argument("invalid handler response", e))?;
+
+        let payload = obj
+            .get_named_property::<napi::JsUnknown>("payload")
+            .map_err(|e| api::Error::invalid_argument("payload missing in handler response", e))?;
+
+        let extra_headers = obj
+            .get_named_property::<napi::JsUnknown>("extraHeaders")
```

Comment:
Can we make `extraHeaders` optional, and treat it being missing as no headers? And if there is no middleware we don't send it?

Reply:
its optiona in ts now, and `get_named_property` will not fail if it isnt set, but return something that is undefined, that we already handle in the parse functions, so I removed those error messages and replaced them with an internal error